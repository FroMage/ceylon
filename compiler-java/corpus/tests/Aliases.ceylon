class Aliases() {
	
	interface Foo<X1,Y,Z> {}
	interface Bar<X2> {}
	interface Baz<X3> {}
	
	alias Simple satisfies Bar<Baz<String>>;
	
	alias WithTypeParameter<X4> satisfies Bar<Baz<X5>>;
	
	alias WithTypeConstraint<X6> satisfies Bar<Baz<X7>>
		where X1 satisfies String;
	
	alias Multiple satisfies Foo<Bar<Baz<String>>, Natural, Character>, Bar<Baz<String>>;
	
	doc "an alias"
	by "Gavin King" "Andrew Haley"
	see #Foo #Bar #Baz
	public alias WithAnnotations satisfies Foo<Bar<Baz<String>>, Natural, Character>;
	
}