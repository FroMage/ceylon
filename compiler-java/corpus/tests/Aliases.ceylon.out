# Aliases.ceylon

(TYPE_DECL
  (TYPE_DECL
    (CLASS_DECL
      (TYPE_NAME Aliases) FORMAL_PARAMETER_LIST
      (BLOCK
        (TYPE_DECL
          (INTERFACE_DECL
            (TYPE_NAME Foo)
            (TYPE_PARAMETER_LIST
              (TYPE_PARAMETER
                (TYPE_NAME X))
              (TYPE_PARAMETER
                (TYPE_NAME Y))
              (TYPE_PARAMETER
                (TYPE_NAME Z)))))
        (TYPE_DECL
          (INTERFACE_DECL
            (TYPE_NAME Bar)
            (TYPE_PARAMETER_LIST
              (TYPE_PARAMETER
                (TYPE_NAME X)))))
        (TYPE_DECL
          (INTERFACE_DECL
            (TYPE_NAME Baz)
            (TYPE_PARAMETER_LIST
              (TYPE_PARAMETER
                (TYPE_NAME X)))))
        (TYPE_DECL
          (ALIAS_DECL
            (TYPE_NAME Simple)
            (SATISFIES_LIST
              (TYPE
                (TYPE_NAME Bar)
                (TYPE_ARG_LIST
                  (TYPE
                    (TYPE_NAME Baz)
                    (TYPE_ARG_LIST
                      (TYPE
                        (TYPE_NAME String)))))))))
        (TYPE_DECL
          (ALIAS_DECL
            (TYPE_NAME WithTypeParameter)
            (TYPE_PARAMETER_LIST
              (TYPE_PARAMETER
                (TYPE_NAME X)))
            (SATISFIES_LIST
              (TYPE
                (TYPE_NAME Bar)
                (TYPE_ARG_LIST
                  (TYPE
                    (TYPE_NAME Baz)
                    (TYPE_ARG_LIST
                      (TYPE
                        (TYPE_NAME X)))))))))
        (TYPE_DECL
          (ALIAS_DECL
            (TYPE_NAME WithTypeConstraint)
            (TYPE_PARAMETER_LIST
              (TYPE_PARAMETER
                (TYPE_NAME X)))
            (SATISFIES_LIST
              (TYPE
                (TYPE_NAME Bar)
                (TYPE_ARG_LIST
                  (TYPE
                    (TYPE_NAME Baz)
                    (TYPE_ARG_LIST
                      (TYPE
                        (TYPE_NAME X)))))))
            (TYPE_CONSTRAINT_LIST
              (TYPE_CONSTRAINT
                (TYPE_NAME X)
                (SATISFIES_LIST
                  (TYPE
                    (TYPE_NAME String)))))))
        (TYPE_DECL
          (ALIAS_DECL
            (TYPE_NAME Multiple)
            (SATISFIES_LIST
              (TYPE
                (TYPE_NAME Foo)
                (TYPE_ARG_LIST
                  (TYPE
                    (TYPE_NAME Bar)
                    (TYPE_ARG_LIST
                      (TYPE
                        (TYPE_NAME Baz)
                        (TYPE_ARG_LIST
                          (TYPE
                            (TYPE_NAME String))))))
                  (TYPE
                    (TYPE_NAME Natural))
                  (TYPE
                    (TYPE_NAME Character))))
              (TYPE
                (TYPE_NAME Bar)
                (TYPE_ARG_LIST
                  (TYPE
                    (TYPE_NAME Baz)
                    (TYPE_ARG_LIST
                      (TYPE
                        (TYPE_NAME String)))))))))
        (TYPE_DECL
          (ALIAS_DECL
            (TYPE_NAME WithAnnotations)
            (SATISFIES_LIST
              (TYPE
                (TYPE_NAME Foo)
                (TYPE_ARG_LIST
                  (TYPE
                    (TYPE_NAME Bar)
                    (TYPE_ARG_LIST
                      (TYPE
                        (TYPE_NAME Baz)
                        (TYPE_ARG_LIST
                          (TYPE
                            (TYPE_NAME String))))))
                  (TYPE
                    (TYPE_NAME Natural))
                  (TYPE
                    (TYPE_NAME Character))))))
          (ANNOTATION_LIST
            (USER_ANNOTATION
              (ANNOTATION_NAME doc)
              (STRING_CST "an alias"))
            (USER_ANNOTATION
              (ANNOTATION_NAME by)
              (STRING_CST "Andrew Haley"))
            (USER_ANNOTATION
              (ANNOTATION_NAME see))
            (LANG_ANNOTATION public)))))))
